### PR TITLE
docs: macOS install steps for cilium cli used incorrect brace expansion

### LINKED ISSUE #31257: Documentation: Incorrect brace expansion in curl command for Cilium CLI installation on macOS
## Cilium Documentation Update

**Description:**
The [current installation instructions](https://docs.cilium.io/en/stable/gettingstarted/k8s-install-default/#install-the-cilium-cli) for the Cilium CLI on macOS contain a curl command that uses brace expansion to download both the tarball and its corresponding SHA256 checksum file. However, the brace expansion is not performed correctly due to the braces being enclosed within double quotes. This results in a malformed URL and a 404 error when attempting to download the files.

**Affected Command:**
`curl -L --fail --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-darwin-${CLI_ARCH}.tar.gz{,.sha256sum}
`

**Issue:**
The brace expansion {,.sha256sum} is intended to download both the cilium-darwin-${CLI_ARCH}.tar.gz file and its checksum file cilium-darwin-${CLI_ARCH}.tar.gz.sha256sum. However, the expansion does not occur because the braces are inside the double quotes, which prevents the shell from interpreting them as intended.

**Expected Behavior:**
The curl command should download both the tarball and the SHA256 checksum file.

**Actual Behavior:**
The curl command fails with a 404 error because the URL is malformed due to the incorrect brace expansion.

**Solution:**
Move the braces outside of the double quotes to allow the shell to perform the brace expansion correctly.

**Impact:**
Users following the current installation instructions on macOS will encounter a 404 error and will be unable to download the Cilium CLI and its checksum file. This issue affects the usability and accessibility of the installation process.

### PR DESCRIPTION
**Pull request by sethstuart**
## Documentation: Incorrect brace expansion in curl command for Cilium CLI installation on macOS

**Description:**
This pull request addresses an issue with the Cilium CLI installation instructions for macOS in the [Getting Started Guide](https://docs.cilium.io/en/stable/gettingstarted/k8s-install-default/#install-the-cilium-cli). The `curl` command for downloading the Cilium CLI and its SHA256 checksum file used incorrect brace expansion, resulting in a 404 error.

**Issue:**
The original command was:
`curl -L --fail --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-darwin-${CLI_ARCH}.tar.gz{,.sha256sum}`

The braces `{}` were inside the double quotes, preventing the shell from performing the brace expansion correctly.

**Solution:**
I have corrected the command by moving the braces outside of the double quotes:
`curl -L --fail --remote-name-all "https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-darwin-${CLI_ARCH}.tar.gz"{,.sha256sum}`

This change allows the shell to perform the brace expansion correctly, enabling the download of both the tarball and the SHA256 checksum file.

**Testing:**
I have tested the corrected command on my macOS system running zsh, and it successfully downloads the required files without any 404 errors.

**Additional QOL Changes:**
- Bumped the version number for cilium cli in the warning above the install steps from `v0.15.0` to `v0.16.0`
- Added a small section to demonstrate how to validate the cilium cli install
- Added the `cilium version --client` command to the installation steps, allowing users to immediately verify the installed version of the Cilium CLI.

Fixes: [#31257](https://github.com/cilium/cilium/issues/31257)

```release-note
Fixed incorrect brace expansion in the curl command for Cilium CLI installation instructions for macOS. Added validation steps and immediate version check to the installation process.
```

Signed-off-by: Seth Stuart <joshuastuart24@gmail.com>

### CODE REVIEW

**Comment by sethstuart** [comment]
This would likely need the label for kind/bug. If that can be added I would appreciate it.

File: Documentation/installation/cli-download.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .. warning::
-  Make sure you install `cilium-cli v0.15.0 <https://github.com/cilium/cilium-cli/releases/tag/v0.15.0>`_
+  Make sure you install `cilium-cli v0.16.0 <https://github.com/cilium/cilium-cli/releases/tag/v0.16.0>`_
```
Comment (ghost) [inline Documentation/installation/cli-download.rst:2]: I didn't understand why the version bump here. Do the instructions not work with Cilium CLI v0.15? If they do, then it should stay "v0.15.0 or later" I think.

File: Documentation/installation/cli-download.rst [hunk 2]
```diff
@@ -52,3 +54,19 @@ various features (e.g. clustermesh, Hubble).
...
+
+.. code-block:: shell-session
+
+   cilium version --client
+
+You should see output similar to the following:
+
+.. code-block:: shell-session
+
+   cilium-cli:  compiled with go1.22.0 on linux/amd64
```
Comment (ghost) [inline Documentation/installation/cli-download.rst:68]: Huh, this looks like a Cilium CLI bug - I believe the Cilium CLI version should be printed before "compiled".

Could you report this bug in cilium/cilium-cli?

IMO this example output should display the Cilium CLI version even if it's not printed by some versions of Cilium CLI.

**Review by ghost — CHANGES_REQUESTED** [review]
Thank you for the PR @sethstuart! Looks good overall. I left one question and it seems that there is a bug in the Cilium CLI output.

Could you also rebase the PR with the main branch instead of merging it?

File: Documentation/installation/cli-download.rst (hunk 2 above)
Comment (julianwiedmann) [reply Documentation/installation/cli-download.rst:68]: This was https://github.com/cilium/cilium-cli/issues/2395. With a fixed release, the output should have the CLI version again. Let's adjust here accordingly.

**Comment by lmb** [comment]
Marking this as draft since there hasn't been progress for a while. Please mark as ready once the last round of feedback has been addressed.

**Pull request closed without merging**